custom/spi.rst

```diff
@@ -11,21 +11,23 @@ them publish values.
 Please refer to the SPI library docs for more information.
 
 .. code-block:: cpp
-
-    #include "esphome.h"
-
-    class MyCustomComponent : public Component {
-     public:
-      void setup() override {
-        SPI.pins(SCK_PIN, MISO_PIN, MOSI_PIN, CS_PIN);
-        SPI.begin();
-      }
-      void loop() override {
-        SPI.beginTransaction(...)
-
-        SPI.write(0x42);
-      }
-    };
+#include "esphome.h"
+
+
+class MyCustomComponent : public Component, public spi::SPIDevice<spi::BIT_ORDER_MSB_FIRST, spi::CLOCK_POLARITY_LOW,
+                                                 spi::CLOCK_PHASE_LEADING, spi::DATA_RATE_2MHZ>{
+ public:
+  void setup() override {
+    
```

Comment:
I'm not a fan of the way in which the methods are formatted here.
I would suggest to have a white line between `setup()` and `loop()` and get rid of the empty line at the start of the method. That makes it a lot more clear what goes together.